Build.fsx

```diff
@@ -18,28 +18,47 @@ let nuGetPackages = !! (nuGetOutputFolder @@ "*.nupkg" )
 let signKeyPath = FullName "Src/AutoFixture.snk"
 let solutionsToBuild = !! "Src/All.sln"
 let processorArchitecture = environVar "PROCESSOR_ARCHITECTURE"
-
-type BuildVersionInfo = { assemblyVersion:string; fileVersion:string; infoVersion:string; nugetVersion:string }
-let calculateVersionFromGit buildNumber =
+// === vNext config ===
+// To disable vNext branch set all the variables below to None
+let vNextBranchName = Some "v4"
+// Specify sha1 of the master branch commit the vNext branch was forked from
+let vNextForkedCommitSha = Some "a177824f05e911c70aac1a459b8da5bd206cf410"
```

Comment:
@moodmosaic If you forget to update variable (and it will be `None`), version will not be updated to  `4.0.0-alpha.X`. I'll improve that later by creation of a record, if we agree on approach with Enrico.

@ecampidoglio I know about that command, but I don't see how it could help. I use this hash to calculate number of commits in the fork branch and [generate](https://github.com/AutoFixture/AutoFixture/blob/ee2214c05294cd72565ea6c7442cbce8e1d1f7d3/Build.fsx#L98) __unique__ and __steady__ build version. As you noticed, the `merge-base` output changes after each merge from `master`, so the generated build version will not be unique and steady.

Instead I looked at the `git rev-list --count HEAD ^master` command to generate constantly increasing numbers. However, this approach [has drawbacks](http://stackoverflow.com/questions/1527234/finding-a-branch-point-with-git#comment13238893_10174692) and isn't reliable because of them, so I decided to not use it.

If you still see that the approach you suggested is suitable, please describe the whole version generation algorithm - likely it differs from mine.

P.S. I've already reviewed a lot of approaches to calculate this commit automatically (like ones mentioned [here](http://stackoverflow.com/questions/1527234/finding-a-branch-point-with-git)), but none of them is suitable and works reliably for our repo.